app/editor/src/hooks/api-editor/useApi.ts

```diff
@@ -0,0 +1,61 @@
+import { AxiosInstance, AxiosRequestConfig } from 'axios';
+import React from 'react';
+import { useAppStore } from 'store/slices';
+import { defaultEnvelope, LifecycleToasts, useSummon } from 'tno-core';
+
+import { Settings } from '.';
+
+/**
+ * Common hook to make requests to the PIMS APi.
+ * @returns CustomAxios object setup for the PIMS API.
+ */
+export const useApi = (
```

Comment:
Shared root hook for api endpoints.  This one wraps and will handle errors.  Also note it provides us a way to determine if something is currently being loaded by an AJAX request.